lib/matplotlib/animation.py

```diff
@@ -91,7 +92,66 @@ def __getitem__(self, name):
 writers = MovieWriterRegistry()
 
 
-class MovieWriter(object):
+class AbstractMovieWriter(six.with_metaclass(abc.ABCMeta)):
+    '''
+    Abstract base class for writing movies. Fundamentally, what a MovieWriter
+    does is provide is a way to grab frames by calling grab_frame().
+
+    setup() is called to start the process and finish() is called afterwards.
+
+    This class is set up to provide for writing movie frame data to a pipe.
+    saving() is provided as a context manager to facilitate this process as::
+
+      with moviewriter.saving('myfile.mp4'):
```

Comment:
Ah, right.  The docstrings are only slightly edited versions of the original docstrings from the `MovieWriter` class.  Copy-edit suggestions are welcome.

After looking at this again, I now think the signature of the `saving` method in the abc should be `def saving(self, fig, outfile, dpi, *args)`.  This API is based on how a writer is used in the `saving` method of the `Animation` class.  There it is only called as `with writer.saving(self._fig, filename, dpi):`.  Even `*args` could be dropped from the signature.